### PR TITLE
Added support for base64 encoded remote config, fixes #162

### LINKED ISSUE #62: [help wanted] How to choose the node of the proxy
Thank you very much for creating this tool. I found it very useful, but I have a question: I don't know how to choose the node of the proxy.

### LINKED ISSUE #162: [Subscription] Support decoding after fetching config.
**Is your feature request related to a problem? Please describe.**

My subscription links returns a base64 encoded file, rather than plain text. This is causing ```mihoro setup```, or ```mihomo test``` to fail.

**Describe the solution you'd like**

Add a config entry like ```remote_config_encoding``` or something. If it's base64, I can manually set it to base64 and mihoro will decode that for me ,automatically.

**Describe alternatives you've considered**
Currently I'm manually doing ```curl <sub_url> | base64 -d -```.

**Additional context**
```
[pairman@fedora Downloads]$ mihoro setup
mihoro: Setting up mihomo's binary, config, and systemd service...
mihoro: Assuming mihomo binary already installed at /home/pairman/.local/bin/mihomo, skipping setup
download: Downloaded to /home/pairman/.config/mihomo/config.yaml
          00:00:00 - Download speed 0 B/s                                                               
error: invalid type: string "<br /> <b>Warning</b>", expected struct MihomoYamlConfig
[pairman@fedora Downloads]$ mihomo test
FATA[2024-07-21T14:53:05.364926708+08:00] Parse config error: yaml: line 2: mapping values are not allowed in this context 

```

### PR DESCRIPTION
**Pull request by RongDuJiKsp**
### Description

Added support for the configuration item remote_config_encoding. When the configuration item is "plain", it will not be processed. When the configuration item is "base64", it will attempt to base64 decoding of the subscription address.

### Linked Issues

fixes #162

### Additional context

This is my first PR.I tested my code ,but I don't know the workflow.  If I am at fault, please correct me  :)

### CODE REVIEW

**Comment by spencerwooo** [comment]
Thank you for this contribution! I will need some time to review your changes. In the meantime, please adjust according to the checks in the CI. :heart:

File: src/utils.rs [hunk 1]
```diff
@@ -126,3 +127,39 @@ pub fn extract_gzip(gzip_path: &str, filename: &str, prefix: &str) -> Result<()>
...
+
+/// Try and decode a base64 encoded file in place.
+///
+/// Decodes the base64 encoded content of a file in place and writes the decoded content back to the
+/// file. If the file does not contain base64 encoded content, maintains the file as is.
+///
+/// # Arguments
+///
+/// * `filename` - A string slice that holds the path to the file to decode.
+pub fn try_decode_base64_file_inplace(filename: &str) -> Result<()> {
```
Comment (spencerwooo) [inline src/utils.rs:139]: I have updated the base64 file decoding function for better error handling, could you please check for any incorrect implementations or issues that I may have introduced here?

**Review by spencerwooo — CHANGES_REQUESTED** [review]
Considering the base64 file decoding fails silently and keeps the original file intact, I would prefer removing the requirement for enabling the decoding via `remote_config_encoding` configuration option, and always apply the base64 decoding regardless of the file format.

Could you please remove any related logic and update this PR? Thanks. :heart:

**Review by spencerwooo — APPROVED** [review]

**Comment by spencerwooo** [comment]
Thanks again for the effort 🤩

**Comment by RongDuJiKsp** [comment]
❤️

**Pull request merged**